packages/curl.rb

```diff
@@ -3,108 +3,237 @@
-     armv7l: '0273d3cf0ffb3d18777bb4d5da6a20c00970ad3787687e48c023f406aa42d4bc',
-       i686: 'ead35292354673e20806e70a9359306bd63758a04db3888eb5eca7c99cd1904a',
-     x86_64: '81b8641d60b222248a75f910e2bf82ce7b9d793a84558466e7670215bbd9b8d1'
+    aarch64: '268771c547ef80761b937aedafe786ca701524e20b4737263aee9791514189d0',
+     armv7l: '268771c547ef80761b937aedafe786ca701524e20b4737263aee9791514189d0',
+       i686: '45796384ef70732ab7d4df71f09d2860e22bee776f86f7ff60c9863ff5bc28c2',
+     x86_64: 'ee99c0558b9dc9e64e4a2619706224daa3e7a0d4b100af135ac5b0a6a3b68ab4'
   })
 
-  depends_on 'brotli' # R
-  depends_on 'ca_certificates' # L
-  depends_on 'c_ares' # R
-  depends_on 'groff' => :build
-  depends_on 'libidn2' # R
-  depends_on 'libmetalink' # R
-  depends_on 'libnghttp2' # R
-  depends_on 'libpsl' # R
-  depends_on 'libtirpc' # ?
-  depends_on 'libunbound' # ?
-  depends_on 'openldap' # R
-  depends_on 'openssl' # R
+  depends_on 'ca_certificates' => :build
+  depends_on 'libcurl' # L (This is only here to make sure upgrades of curl pull in libcurl.)
+  depends_on 'libunbound' => :build
+  depends_on 'musl' => :build
   depends_on 'py3_pip' => :build
   depends_on 'rust' => :build
   depends_on 'valgrind' => :build
-  depends_on 'zlibpkg' # R
-  depends_on 'zstd' # R
 
   def self.build
     raise StandardError, 'Please remove libiconv before building.' if File.exist?("#{CREW_LIB_PREFIX}/libcharset.so")
 
-    @krb5_static_libs = "#{CREW_LIB_PREFIX}/libkrb5support.a " + "#{CREW_LIB_PREFIX}/libgssapi_krb5.a " + "#{CREW_LIB_PREFIX}/libkrb5.a " + "#{CREW_LIB_PREFIX}/libk5crypto.a " + "#{CREW_LIB_PREFIX}/libcom_err.a "
+    @krb5_static_libs = '-l:libkrb5support.a -l:libgssapi_krb5.a -l:libkrb5.a -l:libk5crypto.a -l:libcom_err.a'
 
-    FileUtils.mkdir 'ssl'
-    FileUtils.ln_s "#{CREW_PREFIX}/include/openssl", 'ssl/include'
-    FileUtils.ln_s CREW_LIB_PREFIX, 'ssl/lib'
-    @sslpath = "#{`pwd`.chomp}/ssl"
+    FileUtils.mkdir_p 'build'
+    FileUtils.mkdir_p 'deproot/include'
+    FileUtils.mkdir_p 'deproot/lib'
+    Dir.chdir 'deproot' do
+      FileUtils.ln_s 'lib', 'lib64'
+    end
+    @deppath = "#{`pwd`.chomp}/deproot"
+    FileUtils.ln_s "#{CREW_PREFIX}/include/linux", "#{@deppath}/include/"
+    FileUtils.ln_s "#{CREW_PREFIX}/include/asm", "#{@deppath}/include/"
+    FileUtils.ln_s "#{CREW_PREFIX}/include/asm-generic", "#{@deppath}/include/"
+
+    @arch_ssp_cflags = ''
+    case ARCH
+    when 'i686'
+      @arch_ssp_cflags = '-fno-stack-protector'
+    end
+
+    @curldep_cmake_options = "CC=musl-gcc \
+          CFLAGS='-flto -pipe -O3 -fPIC -ffat-lto-objects -fipa-pta -fno-semantic-interposition -fdevirtualize-at-ltrans #{@arch_ssp_cflags}' \
+          CXXFLAGS='-flto -pipe -O3 -fPIC -ffat-lto-objects -fipa-pta -fno-semantic-interposition -fdevirtualize-at-ltrans #{@arch_ssp_cflags}' \
+          CPPFLAGS='-I#{@deppath}/include' \
+          LDFLAGS='-L#{@deppath}/lib' \
```

Comment:
What's wrong with just using `#{@deppath}/#{ARCH_LIB}`?  Then you don't need any crazy symlink madness.

Reply:
This way it is the same across all architectures. It's only used during build anyways, since these are only built to create the static libraries.

The symlink is needed in any case, because some of these packages are happy to use lib, and some want to use lib64 anyways, but at least this way you don't have to police them to do the right thing.

(same applies to the other similar comments. :) )

Reply:
The whole idea of `#{ARCH_LIB}` is to abstract the library so you don't have to police. :)